Twitter.js

```diff
@@ -33,39 +30,86 @@
+		return "blogPost";
+	} else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
 }
 
-function doWeb(doc, url) { return FW.doWeb(doc, url); }
 
-FW.Scraper({
-itemType         : "blogPost",
-detect           : FW.Xpath('//div[contains(@class,"permalink-tweet-container")]'),
-attachments      : { url : FW.Url() ,
-					 title : "Tweet Snapshot" ,
-					 type : "text/html",
-					 snapshot: true },
-blogTitle		 : FW.Url().match(/([^\/]+)\/status\/\d+/,1).prepend("@"),
-creators         : FW.Xpath('//div[contains(@class,"permalink-header")]//strong[contains(@class,"fullname")]')
-						.text().cleanAuthor("author"),
-// This part is fragile, since we have to iterate over the attributes
-date         : FW.Xpath('//div[contains(@class,"permalink-tweet-container")]//span[@class="metadata"]/span[1]')
-						.text(),
-websiteType		 : "microblog",
-title			 : FW.Xpath('//div[contains(@class,"permalink-tweet-container")]//p[contains(@class, "js-tweet-text")]').text(),
-hooks            : {
-		"scraperDone" : function (item, doc, url) {
-				var match = item.date.match(/(\d+:\d+(?: AM| PM)?) - (\d+) ([A-Za-z]+) (\d+)/);
-				if (match) {
-					item.date = (new Date(match[4] + ' ' + match[3] + ' ' + match[2] + ' ' + match[1])).toGMTString();
-				}
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//div[contains(@class, "content")]');
+	for (var i=0; i<rows.length; i++) {
+		var href = ZU.xpathText(rows[i], './/a[contains(@class, "js-permalink") and contains(@href, "/status/")]/@href');
+		var title = ZU.xpathText(rows[i], './/div[contains(@class, "js-tweet-text-container")]');
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
 			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var item = new Zotero.Item("blogPost");
+	item.title = ZU.xpathText(doc, '//div[contains(@class,"permalink-tweet-container")]//p[contains(@class, "js-tweet-text")]');
+	var author = ZU.xpathText(doc, '//div[contains(@class,"permalink-header")]//strong[contains(@class,"fullname")]');
+	if (author) {
+		item.creators.push(ZU.cleanAuthor(author, "author"));
+	}
+	var date = ZU.xpathText(doc, '//div[contains(@class,"permalink-tweet-container")]//span[@class="metadata"]/span[1]');
+	if (date) {
+		var m = date.match(/(\d\d:\d\d)\s*-\s*(.*)/);
+		if (m) {
+			item.date = ZU.strToISO(m[2]) + "T" + m[1];
+		} else {
+			item.date = date;
+		}
+	}
+	var urlParts = url.split('/');
+	item.blogTitle = '@' + urlParts[3];
+	item.websiteType = "microblog";
```

Comment:
We should maybe reconsider this -- no one seems to actually use it. We could just say "Tweet" which is what APA uses (http://blog.apastyle.org/apastyle/2013/10/how-to-cite-social-media-in-apa-style.html ) and what MLA used to use ( https://www.theatlantic.com/technology/archive/2012/03/how-do-you-cite-a-tweet-in-an-academic-paper/253932/ )